queries.sql

```diff
@@ -30,5 +30,78 @@ WHERE name != 'Gabumon';
 /*Query that Finds all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)*/
 SELECT * FROM animals
 WHERE weight_kg BETWEEN 10.4 AND 17.3;
+/*Query for Rolling Back the species column changes*/
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+
+
+
+/*Update the animals table by setting the species column to "digimon" for all animals that have a name ending in "mon":*/
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+/*UUpdate the animals table by setting the species column to "pokemon" for all animals that don't have species already set::*/
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+COMMIT;
+
+
+/*Removing and bringing back all the records with help of transaction::*/
+BEGIN;
+TRUNCATE TABLE animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+/*Using save point to undo changes done after it have been created*/
+
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+/*Returning number of animals which are there   [THEY ARE 10 ANIMALS IN THE TABLE(10)]*/
+SELECT COUNT(*) FROM animals;
+
+/*Returning number of animals which haven't tried to escape   [THEY ARE 2 ANIMALS WHICH HAVE NOT ESCAPED(2)]*/
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+/*showing avarage weight of animals*/
+SELECT AVG(weight_kg) FROM animals;
+
+/*showing avarage weight of animals*/
+SELECT AVG(weight_kg) FROM animals;
+
+/* Query to determine who escapes the most between neutered and non neutered animals in the table of animals*/
+SELECT neutered, SUM(escape_attempts) AS total_escapes
+FROM animals
+GROUP BY neutered;
+
+/* showing max and min numbers of weights between two species of animals*/
+SELECT species, max(weight_kg), min(weight_kg) from animals GROUP BY species;
+
+
+/* showing average of escape_attempts of type of animals which born  between 1900 and 200*/
+SELECT species, AVG(escape_attempts)  from animals WHERE date_of_birth BETWEEN '1900-01-01' AND '2000-01-01;'
```

Comment:
- [ ] Good job working on this query, but please make sure it is between `1990` and `2000`. At the moment, your query is within `1900` .
Hint:
`BETWEEN '1990-01-01' AND '2000-12-31;`